gurklang/plugins/manim_visualization.py

```diff
@@ -0,0 +1,264 @@
+"""
+Visualizing execution with manim
+
+Make sure you install `manimce`, see instructions at
+https://docs.manim.community/en/v0.1.1/installation.html
+
+Note: manim seems to be slower on PyPy, so run this in CPython.
+
+Scroll to the bottom to see a usage example
+"""
+
+import re
+from itertools import zip_longest
+from functools import reduce
+from typing import Iterator, List,  Tuple, Type
+
+try:
+    import manim
+    from manim import *  # type: ignore
+except ImportError:
+    manim = None
+
+from gurklang.vm import run_with_middleware
+from gurklang.types import Value, Stack, Instruction, Code
+from gurklang.parser import parse
+
+
+def _escape_latex(text: str) -> str:
+    replacements = (
+        ('\\', R'\textbackslash '),
+        ('&', R'\&'),
+        ('%', R'\%'),
+        ('$', R'\$'),
+        ('#', R'\#'),
+        ('_', R'\_'),
+        ('{', R'\{'),
+        ('}', R'\}'),
+        ('~', R'\textasciitilde{}'),
+        ('^', R'\^{}'),
+        ('<', R'\textless{}'),
+        ('>', R'\textgreater{}'),
+    )
+
+    def single_replacement(source: str, replacement: Tuple[str, str]):
+        old, new = replacement
+        return source.replace(old, new)
+
+    return reduce(single_replacement, replacements, text)
```

Comment:
This could've been replaced by
```py
_escape_latex = partial(reduce, single_replacement, replacements)
```
but I decided that it's a little bit confusing